soc/silabs/silabs_s2/xg24/Kconfig.soc

```diff
@@ -23,17 +37,32 @@ config SOC_PART_NUMBER_EFR32MG24B210F1536IM48
 	bool
 	select SOC_SERIES_EFR32MG24
 
+config SOC_PART_NUMBER_MGM240SD22VNA
+	bool
+	select SOC_SERIES_MGM24
+	select SILABS_DEVICE_IS_MODULE
```

Comment:
I don't believe `Kconfig.soc` should reference symbols outside of the `SOC_` namespace (the `Kconfig.soc` files should be possible to use without including any `Kconfig` files). This should likely be done in the normal `Kconfig` instead (as part of all the selections for `SOC_SERIES_MGM24`).